qcodes/widgets/widgets.py

```diff
@@ -197,6 +202,18 @@ def update_widget(self):
                 "loop_indices_label"
             ].value = f"Loop {loop_indices} of {loop_shape}"
 
+            start_time = datetime.strptime(qc.active_dataset().metadata['t_start'], '%Y-%m-%d %H:%M:%S')
+            time_elapsed = datetime.now() - start_time
+            remaining_time = time_elapsed * (1/(qc.active_dataset().fraction_complete() + 1e-6) - 1)
+            estimated_finish_time = start_time + time_elapsed + remaining_time
+            self.widgets[
+                "estimated_completion_time"
+            ].value = f"Estimated completion time: {datetime.strftime(estimated_finish_time, '%Y-%m-%d %H:%M')}"
+            self.widgets[
+                "remaining_time"
+            ].value = f"Remaining time: {int(remaining_time.total_seconds() / 3600)} hours, " \
+                      f"{remaining_time.total_seconds() / 60:.1f} minutes"
```

Comment:
is the plan to show the full remaining time both in hours and in minutes or combined, meaning - 2.1 hours, 126 minutes or 2 hours 6 minutes?

Reply:
The latter.

Reply:
but then you don't subtract the number of minutes counted in hours, since you use the same remaining_time.total_seconds()